crates/ide_completion/src/render/function.rs

```diff
@@ -55,6 +55,19 @@ impl<'a> FunctionRender<'a> {
             .add_call_parens(self.ctx.completion, self.name, params)
             .add_import(import_to_add);
 
+        let mut relevance = CompletionRelevance::default();
+        if let Some(expected_type) = &self.ctx.completion.expected_type {
+            if &self.func.ret_type(self.ctx.db()) == expected_type {
+                relevance.exact_type_match = true;
+            }
```

Comment:
```suggestion
            relevance.exact_type_match = &self.func.ret_type(self.ctx.db()) == expected_type ;
```

Alternatively, we maybe need `item.exact_type_match()` builder method?

Reply:
Updated per your suggestion. As for the builder, I have something else I've been testing that should remove the necessity to write this same relevance code in multiple places. I should have a PR up later today for review.